Regression: After adding rust module, fails to compile on arm64 linux

### Steps To Reproduce

Run `npm install` on cloned master tree on an arm64 machine.

### Expected Result

Succeeds

### Actual Result

```
[  266s] npm ERR! code 1
[  266s] npm ERR! path /home/abuild/rpmbuild/BUILD/desktop-6b6666cd0d289764d708fb3d31ac827bd329bd87/desktop_native
[  266s] npm ERR! command failed
[  266s] npm ERR! command sh -c npm run build-release
[  266s] npm ERR! > desktop_native@0.1.0 build-release
[  266s] npm ERR! > npm run build -- --release
[  266s] npm ERR! 
[  266s] npm ERR! 
[  266s] npm ERR! > desktop_native@0.1.0 build
[  266s] npm ERR! > napi build --release --js true "--release"
[  266s] npm ERR! 
[  266s] npm ERR! Internal Error: Command failed: cargo build --release --target aarch64-unknown-linux-gnu
[  266s] npm ERR!     at checkExecSyncError (node:child_process:828:11)
[  266s] npm ERR!     at Object.execSync (node:child_process:902:15)
[  266s] npm ERR!     at BuildCommand.<anonymous> (/home/abuild/rpmbuild/BUILD/desktop-6b6666cd0d289764d708fb3d31ac827bd329bd87/node_modules/@napi-rs/cli/scripts/index.js:14602:26)
[  266s] npm ERR!     at Generator.next (<anonymous>)
[  266s] npm ERR!     at fulfilled (/home/abuild/rpmbuild/BUILD/desktop-6b6666cd0d289764d708fb3d31ac827bd329bd87/node_modules/@napi-rs/cli/scripts/index.js:3423:58)
[  266s] npm ERR! Compiling serde v1.0.136
[  266s] npm ERR!    Compiling proc-macro2 v1.0.36
[  266s] npm ERR!    Compiling unicode-xid v0.2.2
[  266s] npm ERR!    Compiling heck v0.4.0
[  266s] npm ERR!    Compiling smallvec v1.8.0
[  266s] npm ERR!    Compiling syn v1.0.86
[  266s] npm ERR!    Compiling libc v0.2.119
[  266s] npm ERR!    Compiling pkg-config v0.3.24
[  266s] npm ERR!    Compiling version-compare v0.1.0
[  266s] npm ERR!    Compiling memchr v2.4.1
[  266s] npm ERR!    Compiling version_check v0.9.4
[  266s] npm ERR!    Compiling futures-core v0.3.21
[  266s] npm ERR!    Compiling anyhow v1.0.55
[  266s] npm ERR!    Compiling futures-task v0.3.21
[  266s] npm ERR!    Compiling pin-project-lite v0.2.8
[  266s] npm ERR!    Compiling futures-util v0.3.21
[  266s] npm ERR!    Compiling cfg-if v1.0.0
[  266s] npm ERR!    Compiling pin-utils v0.1.0
[  266s] npm ERR!    Compiling parking_lot_core v0.9.1
[  266s] npm ERR!    Compiling once_cell v1.9.0
[  266s] npm ERR!    Compiling log v0.4.14
[  266s] npm ERR!    Compiling slab v0.4.5
[  266s] npm ERR!    Compiling futures-channel v0.3.21
[  266s] npm ERR!    Compiling scopeguard v1.1.0
[  266s] npm ERR!    Compiling bitflags v1.3.2
[  266s] npm ERR!    Compiling gio v0.15.6
[  266s] npm ERR!    Compiling regex-syntax v0.6.25
[  266s] npm ERR!    Compiling convert_case v0.5.0
[  266s] npm ERR!    Compiling bytes v1.1.0
[  266s] npm ERR!    Compiling napi-build v1.2.1
[  266s] npm ERR!    Compiling futures-io v0.3.21
[  266s] npm ERR!    Compiling lazy_static v1.4.0
[  266s] npm ERR!    Compiling napi-sys v2.1.0
[  266s] npm ERR!    Compiling cfg-expr v0.10.2
[  266s] npm ERR!    Compiling proc-macro-error-attr v1.0.4
[  266s] npm ERR!    Compiling proc-macro-error v1.0.4
[  266s] npm ERR!    Compiling lock_api v0.4.6
[  266s] npm ERR!    Compiling desktop_native v0.0.0 (/home/abuild/rpmbuild/BUILD/desktop-6b6666cd0d289764d708fb3d31ac827bd329bd87/desktop_native)
[  266s] npm ERR!    Compiling quote v1.0.15
[  266s] npm ERR!    Compiling signal-hook-registry v1.4.0
[  266s] npm ERR!    Compiling socket2 v0.4.4
[  266s] npm ERR!    Compiling num_cpus v1.13.1
[  266s] npm ERR!    Compiling aho-corasick v0.7.18
[  266s] npm ERR!    Compiling mio v0.8.0
[  266s] npm ERR!    Compiling parking_lot v0.12.0
[  266s] npm ERR!    Compiling toml v0.5.8
[  266s] npm ERR!    Compiling regex v1.5.4
[  266s] npm ERR!    Compiling futures-executor v0.3.21
[  266s] npm ERR!    Compiling system-deps v6.0.2
[  266s] npm ERR!    Compiling glib-sys v0.15.6
[  266s] npm ERR!    Compiling gobject-sys v0.15.5
[  266s] npm ERR!    Compiling gio-sys v0.15.6
[  266s] npm ERR!    Compiling libsecret-sys v0.1.3
[  266s] npm ERR!    Compiling napi-derive-backend v1.0.25
[  266s] npm ERR!    Compiling thiserror-impl v1.0.30
[  266s] npm ERR!    Compiling tokio-macros v1.7.0
[  266s] npm ERR!    Compiling ctor v0.1.21
[  266s] npm ERR!    Compiling napi-derive v2.1.0
[  266s] npm ERR!    Compiling tokio v1.17.0
[  266s] npm ERR!    Compiling thiserror v1.0.30
[  266s] npm ERR!    Compiling proc-macro-crate v1.1.3
[  266s] npm ERR!    Compiling glib-macros v0.15.6
[  266s] npm ERR!    Compiling glib v0.15.6
[  266s] npm ERR!    Compiling napi v2.1.0
[  266s] npm ERR!    Compiling libsecret v0.1.3
[  266s] npm ERR! error[E0308]: mismatched types
[  266s] npm ERR!     --> /home/abuild/rpmbuild/BUILD/desktop-6b6666cd0d289764d708fb3d31ac827bd329bd87/desktop_native/vendor/libsecret/src/service.rs:1035:17
[  266s] npm ERR!      |
[  266s] npm ERR! 1035 |                 &mut item_paths as *mut _ as *mut *const i8,
[  266s] npm ERR!      |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u8`, found `i8`
[  266s] npm ERR!      |
[  266s] npm ERR!      = note: expected raw pointer `*mut *const u8`
[  266s] npm ERR!                 found raw pointer `*mut *const i8`
[  266s] npm ERR! 
[  266s] npm ERR! For more information about this error, try `rustc --explain E0308`.
[  266s] npm ERR! error: could not compile `libsecret` due to previous error
[  266s] npm ERR! warning: build failed, waiting for other jobs to finish...
[  266s] npm ERR! error: build failed
[  266s] 
[  266s] npm ERR! A complete log of this run can be found in:
[  266s] npm ERR!     /home/abuild/rpmbuild/BUILD/NodeCache/_logs/2022-04-10T19_03_38_432Z-debug-0.log
```
### Screenshots or Videos

I maintain an [unofficial opensuse package for bitwarden](https://build.opensuse.org/package/show/home:dziobian:gulgul-ultron/bitwarden).

See [here](https://gist.github.com/brjsp/e2db447b943c452a00317752e9319d41) for a build log of the package version which fails on aarch64.
[The same source](https://build.opensuse.org/project/show/home:dziobian:branches:home:dziobian:gulgul-ultron) builds correctly on x86_64.



### Additional Context

This seems to be due to `char` being signed on x64 but unsigned on arm64.
Note that this bug [seems already fixed by libsecret upstream](https://gitlab.gnome.org/World/Rust/libsecret-rs/-/commit/0cd3ceb6c5b632decb940a2e3f262dfaab3c6764).



### Operating System

Linux

### Operating System Version

openSUSE Tumbleweed

### Installation method

Other

### Build Version

git commit 6b6666cd0d289764d708fb3d31ac827bd329bd87

--- a/desktop_native/Cargo.toml
+++ b/desktop_native/Cargo.toml
@@ -40,4 +40,4 @@ security-framework-sys = "2.6.1"
 
 [target.'cfg(target_os = "linux")'.dependencies]
 gio = "0.15.6"
-libsecret = "0.1.3"
+libsecret = "0.1.4"

